pkg/operator/ceph/object/controller.go

```diff
@@ -275,15 +275,15 @@ func (r *ReconcileCephObjectStore) reconcile(request reconcile.Request) (reconci
 
 		// Cancel the context to stop monitoring the health of the object store
 		r.stopMonitoring(cephObjectStore)
-
-		cfg := clusterConfig{
-			context:     r.context,
-			store:       cephObjectStore,
-			clusterSpec: r.clusterSpec,
-			clusterInfo: r.clusterInfo,
+		if !cephObjectStore.Spec.IsExternal() {
```

Comment:
Inside the `deleteStore()` method it already checks for external cluster, seems like we don't need to duplicate the check here.